module/Activity/src/Activity/Service/Signup.php

```diff
@@ -220,13 +296,29 @@ public function signOff(ActivityModel $activity, Member $user)
         if (is_null($signUp)) {
             return;
         }
+        $this->removeSignUp($signUp);
+    }
+
+    public function externalSignOff(ExternalActivitySignup $signup)
+    {
+        if (!($this->isAllowed('adminSignup', 'activity') ||
+                $this->isAllowed('adminSignup', $signup->getActivity()))) {
+            $translator = $this->getTranslator();
+            throw new \User\Permissions\NotAllowedException(
+                $translator->translate('You are not allowed to remove external signups for this activity')
+            );
+        }
+        $this->removeSignUp($signup);
+    }
 
+    protected function removeSignUp(ActivitySignup $signup)
+    {
         $em = $this->getServiceManager()->get('Doctrine\ORM\EntityManager');
-        $values = $this->getActivityFieldValueMapper()->getFieldValuesBySignup($signUp);
+        $values = $this->getActivityFieldValueMapper()->getFieldValuesBySignup($signup);
         foreach($values as $value){
             $em->remove($value);
         }
```

Comment:
This for loop shouldn't be needed if `cascade` is set in the entity relation in the models.